metadata-models/src/main/pegasus/com/linkedin/dataset/FieldsLineages.pdl

```diff
@@ -0,0 +1,12 @@
+namespace com.linkedin.dataset
+
+/**
+ * Upstreams of dataset fields.
+ * Key of map is field name in this dataset.
+ */
+
+/*
+Note that this model allows only one lineage per field. 
+If all the lineages induced by multiple writer tasks on a single field are to be captured, this should be done by capturing the lineage induced by each writer task as part of the DataJob corresponding to that task.
+*/
+typeref FieldsLineages = map[string, FieldLineageDetails]
```

Comment:
1. s/FieldsLineages/FieldUpstreamLineages/ ?
2. Modeling this as a map of fieldName ->  upstreamFields is sort of restrictive in the sense that it does not allow us to capture lineages like  (set of upstream fields) -> (set of downstream fields), such as snowflake does, as a first-class lineage information. cc: @shirshanka